### PR TITLE
Thanos dockerfile

### PR DESCRIPTION
**Pull request by ChrisScottThomas**
Creates Alpine Dockerfile from a pinned version, currently latest release.

The entrypoint script creates an example bucket.yml, to be overwritten, and will run a different Thanos job depending on its `mode` set as an env var.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    --data-dir=/local/state/data/dir \
+    --objstore.config-file=bucket.yml
+else
+    /bin/thanos sidecar \
+    --http-address=0.0.0.0:10902 \
+    --grpc-address=0.0.0.0:10901 \
+    --tsdb.path=/path/to/prometheus/data/dir \
+    --prometheus.url=http://localhost:9090 \
+    --objstore.config-file=bucket.yml
+fi
```
Comment (iw-gtadam) [inline entrypoint.sh]: Needs newline

File: entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    
+    elif [ $THANOS_MODE == "store" ]; then
+    /bin/thanos store \
+    --data-dir=/local/state/data/dir \
+    --objstore.config-file=bucket.yml
+else
+    /bin/thanos sidecar \
+    --http-address=0.0.0.0:10902 \
+    --grpc-address=0.0.0.0:10901 \
+    --tsdb.path=/path/to/prometheus/data/dir \
```
Comment (iw-gtadam) [inline entrypoint.sh]: Should tsdb.path this not be to an actual path

Comment (ChrisScottThomas) [reply entrypoint.sh]: it should.

Comment (ChrisScottThomas) [reply entrypoint.sh]: it now, is.

**Review by iw-gtadam — APPROVED** [review]

**Pull request merged**